src_c/imageext.c

```diff
@@ -775,15 +775,9 @@ image_save_ext(PyObject *self, PyObject *arg)
               (name[namelen - 2] == 'p' || name[namelen - 2] == 'P') &&
               (name[namelen - 3] == 'j' || name[namelen - 3] == 'J')))) {
 #ifdef JPEGLIB_H
-            /* jpg save functions seem *NOT* thread safe at least on windows.
-             */
-            /*
-            Py_BEGIN_ALLOW_THREADS;
-            */
+            Py_BEGIN_ALLOW_THREADS; /* safe as long as Python is never used */
```

Comment:
Quite some years ago (10yrs) jpeg saving multiple jpegs at once had problems in threads when releasing the GIL. I never investigated why. It could have been a libjpeg thing at the time... not sure.

Reply:
Holding the GIL blocks multiple threads from using SaveJPEG() simultaneously, so I guess libjpeg wasn't thread safe.